tt_metal/hw/inc/debug/sanitize_noc.h

```diff
@@ -51,17 +49,94 @@ typedef bool debug_sanitize_noc_dir_t;
 #define DEBUG_SANITIZE_NOC_UNICAST false
 typedef bool debug_sanitize_noc_cast_t;
 
+// Helper function to get the core type from noc coords.
+int get_core_type(uint8_t noc_id, uint32_t x, uint32_t y) {
+    core_info_msg_t tt_l1_ptr *core_info = GET_MAILBOX_ADDRESS_DEV(core_info);
+
+    for (uint32_t idx = 0; idx < MAX_DRAM_CORES; idx++) {
+        uint16_t core_x = core_info->dram_cores_x[idx];
+        uint16_t core_y = core_info->dram_cores_y[idx];
+        if (core_x == CORE_COORD_INVALID && core_y == CORE_COORD_INVALID) {
```

Comment:
invalid checks aren't needed as they won't match existing coordinates (reduce code size and speed up the comparisons)

Reply:
Removed these in latest commit